Issues with building noah-owp-modular shared library

There seems to be some issues with manually building shared library for noah-owp-modular

## Current behavior
If I run ```cmake -B extern/noah-owp-modular/cmake_build -S extern/noah-owp-modular```, I get:
```
netCDF-Fortran 4.5.4
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/home/shengting.cui/ngen_automated_test/test/ngen/extern/noah-owp-modular/NETCDF_MODULE_DIR
```
However you can still run ```make -C extern/noah-owp-modular/cmake_build``` to compile the code to completion but with the warning that
```
ngen/extern/noah-owp-modular/NETCDF_MODULE_DIR-NOTFOUND’ [-Wmissing-include-dirs]
```

If I try some hacking by running:  ``` cmake -B extern/noah-owp-modular/cmake_build -S extern/noah-owp-modular -DNGEN_IS_MAIN_PROJECT=ON```, the first step of build is successful. However, in the compile step:
```make -C extern/noah-owp-modular/cmake_build```, it generates the following error:
```
ngen/extern/noah-owp-modular/noah-owp-modular/bmi/bmi_noahowp.f90:6:7:

    use bmif_2_0_iso
       1
Fatal Error: Can't open module file ‘bmif_2_0_iso.mod’ for reading at (1): No such file or directory
compilation terminated.
```


## Expected behavior
Both build steps run successfully without warning and generate a share library ```extern/noah-owp-modular/cmake_build/libsurfacebmi.so```

## Steps to replicate behavior (include URLs)

As described above.


## Screenshots

--- a/extern/noah-owp-modular/CMakeLists.txt
+++ b/extern/noah-owp-modular/CMakeLists.txt
@@ -28,6 +28,16 @@ target_include_directories(surfacebmi INTERFACE "${_SURFACEBMI_BINARY_DIR}/mod")
 target_compile_definitions(surfacebmi PRIVATE BMI_ACTIVE)
 
 if(NGEN_IS_MAIN_PROJECT)
+
+    # This ensures we can build NOAH-OWP-Modular with NGen support, but
+    # separate from NGen.
+    if(NOT TARGET iso_c_bmi)
+        add_subdirectory(
+            "${CMAKE_CURRENT_LIST_DIR}/../iso_c_fortran_bmi"
+            "${CMAKE_CURRENT_BINARY_DIR}/iso_c_fortran_bmi"
+        )
+    endif()
+
     target_link_libraries(surfacebmi PUBLIC iso_c_bmi)
     target_compile_definitions(surfacebmi PRIVATE NGEN_FORCING_ACTIVE NGEN_OUTPUT_ACTIVE NGEN_ACTIVE)
 else()

